core/shared/src/main/scala/scalaz/zio/package.scala

```diff
@@ -22,9 +22,10 @@ package object zio extends EitherCompat {
   type Canceler = UIO[_]
   type FiberId  = Long
 
-  type IO[+E, +A] = ZIO[Any, E, A]
-  type Task[+A]   = ZIO[Any, Throwable, A]
-  type UIO[+A]    = ZIO[Any, Nothing, A]
+  type IO[+E, +A]    = ZIO[Any, E, A]
+  type Task[+A]      = ZIO[Any, Throwable, A]
+  type TaskR[-R, +A] = ZIO[R, Throwable, A]
```

Comment:
@jdegoes Something like this?

```scala
object TaskR extends ZIO_E_Throwable {
  type UpperE = Throwable

  def apply[R, A](a: => A): TaskR[R, A] = effect(a)
}
```

Or this?

```scala
object TaskR extends ZIO_E_Throwable {
  type UpperE = Throwable

  def apply[A](a: => A): Task[A] = effect(a)
}
```

I'm confused by the `IO` object that returns a `Task`.

Reply:
An `IO` object can return a task because a task can always be interpreted as an `IO` for some choice of `E` and `A`. The reverse is not true.

So the latter one is correct because a `Task` can always be interpreted as a `Task`. The former one is also correct but contains a useless `R` parameter that will only cause type inference problems (Scala can't infer it).